v2/stack_analyses.yaml

```diff
@@ -0,0 +1,455 @@
+openapi: 3.0.1
+info:
+  title: Stack Analyses API
+  version: "2.0"
+servers:
+  - url: https://f8a-analytics-2445582058137.{environment}.gw.apicast.io/api/v2
+    variables:
+      environment:
+        default: api    # Production server
+        enum:
+          - production # Production server
+          - staging # Staging server
+security:
+  - tokenAuth: []
+paths:
+  /stack-analyses:
+    post:
+      tags:
+      - Scan Services
+      summary: Initiates a scan to gather security and other details about an application
+        stack and related recommendations to improve the stack
+      operationId: f8a_scanner.api_v2.stack-analyses-post
+      parameters:
+      - name: ecosystem
+        in: header
+        description: The ecosystem of the stack analysis being called.
+        required: true
+        schema:
+          type: string
+      - name: origin
+        in: header
+        description: This is required to give an indication as to from where is the
+          request getting generated. If from VScode extension, the value which we
+          need to send is 'vscode'.
+        schema:
+          type: string
+          default: vscode
+          enum:
+            - vscode
+      - name: show_transitive
+        in: header
+        description: This is required to enable or disable the transitive support.
+          We need to pass the value as 'true' if transitive is needed. Default value
+          is 'false'.
+        schema:
+          type: boolean
+          default: true
+      requestBody:
+        content:
+          multipart/form-data:
+            schema:
+              required:
+              - filePath[]
+              - manifest[]
+              properties:
+                manifest[]:
+                  type: string
+                  description: A manifest file dependencies.txt for maven, npmlist.json
+                    for npm & pylist.json for pypi. Commands to generate these files
+                    are mentioned in the README.md
+                  format: binary
+                filePath[]:
+                  type: string
+                  description: The absolute or relative path where the manifest file
+                    is located. This has to be mandatorily provided with the manifest[]
+                    parameter and vice versa.
+        required: true
+      responses:
+        200:
```

Comment:
If UUID is generated would UUID be part of the response while stack analysis is triggered ?

Reply:
No, it won't be part of response.

Reply:
If UUID is generated on the server-side (to make sense across all possible integrations), UUID got to be part of the server-side.

Reply:
The separate end point (not public facing, internal) for UUID makes sense.

Reply:
A separate endpoint just to generate uuid at the server side is also fine. We just need to make sure that this endpoint is called at the time of installation and then the uuid is stored locally and is sent for the stack analysis requests. Please dont confuse with my previous comment. These are 2 separate ways of doing the same thing. As long as we dont expect uuid generation at the time of registration (with snyk) i think it should be fine if its generated in the server side via a separate API call.

Reply:
The response header for the POST call (which triggers the stack analysis) would need send back UUID as part of the response header (if no UUID been sent as part of the request)

Reply:
uuid field is mandatory in the response header as well. you can check component analysis for example
cc: @deepak1725